### ISSUE TITLE
enhancement: Maya Pointcache export allow exporting -userAttr instead of -attr

### ISSUE THREAD

**Issue by BigRoy** [ynput/OpenPype#4812]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Please state which of feature you have in mind and describe what are its shortcomings?

The Maya Alembic exporter allows to export custom attributes as geometric attributes or user attributes. So far OpenPype has always exported all custom attributes using `-attr` and `-attrPrefix` flags which result in geometric attributes and it has no functionality to decide whether attributes should be exported as `-userAttr` or `-userAttrPrefix`.

**What is the difference?** 

One is a **'geometric attribute'** and the other is a **'user attribute'**. [It defines where it writes it to in the Alembic file](https://github.com/alembic/alembic/blob/a4b60eadb08500ea8ab7af761ddb3377c5772f30/maya/AbcExport/MayaUtility.h#L235-L241) 

- **attr + attrPrefix** go into **`.geomArbParams`**
- **userAttr + userAttrPrefix** go into **`.userProperties`**

But then the question becomes what does that mean for the resulting Alembic file. How does it behave differently and which should you choose when?

In [OpenPype we've always used `attr` and `attrPrefix`](https://github.com/ynput/OpenPype/blob/f4d22e6ac671245dab4c4f6831ea6f057ca088c6/openpype/hosts/maya/plugins/publish/extract_pointcache.py#L57-L58) and our studio has done that for years too. _For example the `cbId` attribute for nodes is there too which as I read this now makes more sense in `.userProperties` maybe?_

It influences for example [how something comes into Katana](https://github.com/ezequielmastrasso/EZLookdevTools/issues/106#issuecomment-544480777). But still, how do I decide which should go where?

And if we want to be that picky. That'd almost make the `includeUserDefinedAttributes` automatic 'add my custom attributes in the export' more complicated since that can't really decide which attribute should go where, right? @tokejepsen what do you think?

And even more important maybe, how would an artist know what to put where?

**Some more details on the differences**

Here's an explanation on the [intended differences](https://groups.google.com/g/alembic-discussion/c/JEu37K4AE2Q/m/9nhCY6KxEcYJ).

> **`-attr`** stores data under the arbitrary geometry property compound (available via `getArbGeomParams()` on AbcGeom schemas)
> 
> The arbitrary geometry property compound is meant to hold data that you want to communicate to other facilities and applications, especially renderers.  Things like per-vertex color, or extra UV sets are kinds of things you might store in here.
> 
> **`-userAttr`** places the data under the user property compound (available via `getUserProperties()` on AbcGeom schemas)
> 
> The user property compound is meant to hold any extra pipeline specific data to help you customize your workflows.
> There is no expectation that other facilities or applications will preserve this data, although some try to.

---

@fabiaserra described a use case of the differences from his previous studio:

> On any case, it's good to try make the cache exporter interface as thin as possible to the internal API to accommodate for any possible workflows. On my prior studio we exposed most of the Alembic options (as Advanced tabs) because in some cases there was a specific intend to write the attribute as user or compound for specific DCCs or applications expecting it that way

### How would you imagine the enhancement of the feature?

We could either expose `-userAttr` and `-userAttrPrefix` as attributes on the instances explicitly so if its needed one could just put their attributes there. However, that would mean exposing to the artists something that might be a slightly technical differentation of the attributes, but it will be the simplest implementation and most explicit.

Another approach could be that in the project settings we define a list of attributes (or attribute prefixes) that should be `-userAttr` typed. So that if an attribute is in those lists then it'll be passed to `-userAttr` instead of `-attr` in the export command. Less explicit, but potentially more control by the studio admin and less hassle for the artist to manage the differences. Potential benefit to this is that the  `includeUserDefinedAttributes` feature that OpenPype offers to automatically include any custom attribute in the export can use these lists too so that even then it can differentiate between the two.

### Is there any more labels you wish to add?

- [X] I have searched labels and added any

### Additional context:

Originally reported in a comment [here](https://github.com/ynput/OpenPype/pull/4388#discussion_r1158551110) by @fabiaserra 

[cuID:[OP-5570](https://app.clickup.com/t/863gef4uu)]

**Comment by tokejepsen** [comment]
Is this anticipating issues down the line or is there a specific use case problem atm?

**Comment by BigRoy** [comment]
> Is this anticipating issues down the line or is there a specific use case problem atm?

As far as I know this is anticipating an issue down the line. I've solely created this issue to track the feature request by @fabiaserra. He might be able to say more about how urgent this is for a production use case.

**Comment by fabiaserra** [comment]
> > Is this anticipating issues down the line or is there a specific use case problem atm?
> 
> As far as I know this is anticipating an issue down the line. I've solely created this issue to track the feature request by @fabiaserra. He might be able to say more about how urgent this is for a production use case.

Yeah not urgent at all from our side, we aren't using OP in production yet in my current studio and I don't foresee this becoming an issue anytime soon, as @BigRoy, we were using exclusively `-attr` for many years before we discovered about `-uattr` in my prior company and we needed to expose those as required by some internal technology we used, but it's not common. I raised it on the PR as it caught my eye when I saw just `-attr` being exposed and being bit by it in the past. Once I get more tracking on using OP in production I'd like to test whether writing the custom attributes as `-uattr` make the caches lighter and if it's worth the effort.

TL;DR: It's anticipating issues and not urgent

**Comment by BigRoy** [comment]
Exposing either or both has been implemented recently in both OpenPype and AYON.

Here is the relevant AYON PR: https://github.com/ynput/ayon-core/pull/336
